### ISSUE TITLE
Potential.load_state_dict() in training mode not compatible with loading a training model

### ISSUE THREAD

**Issue by MarshallYan** [choderalab/modelforge#264]
Key modification in Potential.load_state_dict() in training mode causing an error.

**Comment by wiederm** [comment]
To clarify: the current workflow is intended for training a model in `training` mode, then saving the state dict and loading into a model that is created in `inference` mode. This is done because during training, the potential is wrapped in a training pytorch-lightning class that handles the boilerplate training code.

The issue you are describing appears when saving a state_dict in training mode and loading it in a new model also in training mode. The anticipated workflow for such a use case was a checkpoint file.
The issue is caused by the custom overloading of the `load_state_dict()` method.

**Comment by wiederm** [comment]
I think this has been resolved
